### PR TITLE
JPN-487 Show entry point if there are more than 3 admins 

### PR DESCRIPTION
**Pull request by kamilkoterba**
https://wikia-inc.atlassian.net/browse/JPN-487

Show entry point if there are more than 3 admins add consts for other limits.

@Wikia/spitfires @perjg @d34th4ck3r

### CODE REVIEW

File: extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php [hunk 1]
```diff
@@ -125,17 +130,20 @@ public function getTopContributorsData() {
...
 		$topAdmins = $this->usersModel->filterGlobalBots(
 			// get all admins who have contributed in the last two years ordered by contributions
-			$this->usersModel->getTopContributors( 10, false, true )
+			$this->usersModel->getTopContributors( self::TOP_ADMINS_LIMIT, false, true )
 		);
+		$topAdminsCount = count( $topAdmins );
 		$topAdminsDetails = $this->getContributorsDetails( $topAdmins );
 
-		$remainingAdminCount = count( $topAdmins ) - 2;
+		$remainingAdminCount = $topAdminsCount > self::TOP_ADMINS_MODULE_LIMIT
```
Comment (adamkarminski) [inline extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php]: I'd move it to a separate helping function and unit test it. What do you think?

Comment (adamkarminski) [inline extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php]: Also, it these data operations seem to fit better in the UsersModel than in the controller.

File: extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php [hunk 2]
```diff
@@ -131,7 +132,7 @@ public function getTopAdminsData() {
 	public function getAllAdminsData() {
 		$allAdmins = $this->usersModel->filterGlobalBots(
 			// get all admins who have contributed in the last two years ordered by contributions
-			$this->usersModel->getTopContributors( 10, false, true )
+			$this->usersModel->getTopContributors( self::TOP_ADMINS_LIMIT, false, true )
```
Comment (adamkarminski) [inline extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php]: Please remove limits from these queries.

File: extensions/wikia/CommunityPage/models/CommunityPageSpecialTopAdminsFormatter.class.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+			'admins' => self::prepareAdminsToShow( $topAdminsCount, $remainingAdminCount, $topAdminsDetails ),
+			'otherAdminCount' => $remainingAdminCount,
+			'haveOtherAdmins' => $remainingAdminCount > 0,
+			'adminCount' => $topAdminsCount,
+		];
+	}
+
+	private function prepareRemainingCount( $topAdminsCount ) {
+		return $topAdminsCount > self::TOP_ADMINS_MODULE_LIMIT
+			? $topAdminsCount - self::TOP_ADMINS_MODULE_LIMIT + 1
```
Comment (adamkarminski) [inline extensions/wikia/CommunityPage/models/CommunityPageSpecialTopAdminsFormatter.class.php]: @Grunny Do we have a convention on using ternary operators? I think we talked about them at some point.

File: extensions/wikia/CommunityPage/models/CommunityPageSpecialTopAdminsFormatter.class.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	 * Returns array with fields to supply topAdmins.mustache template
+	 * If there are more than three admins provided returns first two and the count of remaining ones
+	 * @param array $topAdminsDetails
+	 * @return array
+	 */
+	public static function prepareData( array $topAdminsDetails ) {
+		$topAdminsCount = count( $topAdminsDetails );
+		$remainingAdminCount = self::prepareRemainingCount( $topAdminsCount );
+		return [
+			'admins' => self::prepareAdminsToShow( $topAdminsCount, $remainingAdminCount, $topAdminsDetails ),
```
Comment (adamkarminski) [inline extensions/wikia/CommunityPage/models/CommunityPageSpecialTopAdminsFormatter.class.php]: I think this key could be more descriptive. Maybe something like `selectedAdmins` or `topAdmins`?

File: extensions/wikia/CommunityPage/models/CommunityPageSpecialTopAdminsFormatter.class.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+			'otherAdminCount' => $remainingAdminCount,
+			'haveOtherAdmins' => $remainingAdminCount > 0,
+			'adminCount' => $topAdminsCount,
```
Comment (adamkarminski) [inline extensions/wikia/CommunityPage/models/CommunityPageSpecialTopAdminsFormatter.class.php]: Shouldn't it be `topAdminsCount`?

Also, there's an inconsistency - some keys use the singular form `admin` and some the plural one `admins`. Can we change all of them to follow the same form?

File: extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php [hunk 6]
```diff
@@ -105,23 +110,19 @@ public function getTopContributorsData() {
...
-			'admins' => array_slice( $topAdminsDetails, 0, 2 ),
-			'otherAdminCount' => $remainingAdminCount <= self::DISPLAY_COUNT_LIMIT ? $remainingAdminCount : self::DISPLAY_COUNT_LIMIT,
-			'haveOtherAdmins' => $remainingAdminCount > 0,
-			'adminCount' => count( $topAdmins ),
 			'noAdminText' => $this->msg( 'communitypage-no-admins' )->plain(),
 			'noAdminContactText' => $this->msg( 'communitypage-no-admins-contact' )->plain(),
 			'noAdminHref' => $this->msg( 'communitypage-communitycentral-link' )->inContentLanguage()->text(),
-		] );
+		];
+		$this->response->setData( array_merge($templateMessages, $topAdminsTemplateData) );
```
Comment (adamkarminski) [inline extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php]: Spaces in parentheses

File: extensions/wikia/CommunityPage/models/CommunityPageSpecialTopAdminsFormatter.class.php (hunk 3 above)
Comment (Grunny) [inline extensions/wikia/CommunityPage/models/CommunityPageSpecialTopAdminsFormatter.class.php]: Our [conventions](https://www.mediawiki.org/wiki/Manual:Coding_conventions/PHP#Ternary_operator) for ternary operators in PHP are that they are fine but to prefer `if` if it's a multi-line expression for readability (and as a side note, definitely don't want to nest ternary operators). An `if` block to be slightly more readable here, but it's not a big problem either way for a relatively simple one like this.

**Pull request merged**